src/language-js/print/call-arguments.js

```diff
@@ -43,6 +45,25 @@ function printCallArguments(path, options, print) {
     ];
   }
 
+  // Very simple solution to split a case like this into multiple lines:
+  //
+  // registerReducer(({
+  //   a: { b, c },
+  //   d: { e, f },
+  // }) => combine(b, c, e, f));
+  //
+  // Fails to handle a case with more arguments, like this:
+  //
+  // registerReducer((event, { a: { b, c }, d: { e, f } }) => combine(b, c, e, f))
+  if (
+    args.length === 1 &&
+    args[0].type === "ArrowFunctionExpression" &&
+    args[0].params.length > 0 &&
+    shouldBreakForObjectPattern(args[0].params[0])
```

Comment:
This change will fix the formatting for `registerReducer((event, { a: { b, c }, d: { e, f } }) => combine(b, c, e, f))`:

```suggestion
    args[0].params.some(shouldBreakForObjectPattern)
```

but it is still pretty limited. I hope to improve this further.